arcgis-ios-sdk-samples/Utility Network/Perform valve isolation trace/PerformValveIsolationTraceViewController.swift

```diff
@@ -0,0 +1,261 @@
+    
+    // Constants for creating the default trace configuration.
+    let domainNetworkName = "Pipeline"
+    let tierName = "Pipe Distribution System"
+
+    // Constants for creating the default starting location.
+    let networkSourceName = "Gas Device"
+    let assetGroupName = "Meter"
+    let assetTypeName = "Customer"
+    let globalId = UUID(uuidString: "98A06E95-70BE-43E7-91B7-E34C9D3CB9FF")!
+    
+    /// An array to keep track of all available utility categories under current network definition.
+    lazy var filterBarrierCategories: [AGSUtilityCategory] = []
+    
+    var traceConfiguration: AGSUtilityTraceConfiguration?
+    var selectedCategory: AGSUtilityCategory?
+    var startingLocationElement: AGSUtilityElement!
+    var utilityNetwork: AGSUtilityNetwork!
+    
+    /// The gas line layer ./3 and gas device layer ./0 created from the feature service.
+    var layers: [AGSFeatureLayer] {
+        return [3, 0].map {
+            let featureTable = AGSServiceFeatureTable(url: featureServiceURL.appendingPathComponent("\($0)"))
+            let layer = AGSFeatureLayer(featureTable: featureTable)
+            return layer
+        }
+    }
+    
+    // MARK: Initialize map and utility network
+    
+    /// Create a map.
+    ///
+    /// - Returns: An `AGSMap` object.
+    func makeMap() -> AGSMap {
+        let map = AGSMap(basemap: .streetsNightVector())
+        // Add the utility network feature layers to the map for display.
+        map.operationalLayers.addObjects(from: layers)
+        return map
+    }
+    
+    /// Create trace parameters based on the trace configuration from current utility tier and utility category.
+    ///
+    /// - Returns: An optional `AGSUtilityTraceParameters` object.
+    func makeTraceParameters() -> AGSUtilityTraceParameters? {
+        guard let configuration = traceConfiguration else {
+            self.setStatus(message: "Trace configuration does not exist.")
+            return nil
+        }
+        guard let category = selectedCategory else {
+            // A nil category will let the trace fail.
+            self.setStatus(message: "Category not set for filter barrier.")
+            return nil
+        }
+        // Note: AGSUtilityNetworkAttributeComparison or AGSUtilityCategoryComparison with AGSUtilityCategoryComparisonOperator.doesNotExist
+        // can also be used. These conditions can be joined with either AGSUtilityTraceOrCondition or AGSUtilityTraceAndCondition.
+        let comparison = AGSUtilityCategoryComparison(category: category, comparisonOperator: .exists)
+        configuration.filter?.barriers = comparison
+        configuration.includeIsolatedFeatures = isolationSwitch.isOn
+        
+        let parameters = AGSUtilityTraceParameters(traceType: .isolation, startingLocations: [startingLocationElement])
+        parameters.traceConfiguration = configuration
+        return parameters
+    }
+    
+    func loadUtilityNetwork() {
```

Comment:
This can be put into `viewDidLoad`, but I extract the loading process into a separate function for clearness, as well as to decrease the line count in `viewDidLoad` method.